content/software_envs/containers.md

```diff
@@ -1,42 +1,47 @@
+Currently, two container solutations are supported  **Apptainer** (previously called Singularity), and **Docker**.
 
-### Using Singularity
+### Using Apptainer
 
-The unpacked.cern.ch service mounts on CVMFS contains many singularity images, some of which are suitable for machine
+#### Quickstart
+One-line access to TensorFlow + PyTorch + Numpy + JupyterLab on lxplus:
+```bash
+apptainer shell -B /afs -B /eos --nv /cvmfs/unpacked.cern.ch/registry.hub.docker.com/cmsml/cmsml:latest
+```
+
+#### More information
+
+The unpacked.cern.ch service mounts on CVMFS contains many apptainer images, some of which are suitable for machine
  learning applications.
-A description of each of the images is beyond the scope of this document. 
-However, if you find an image which is useful for your application, you can use if by running a Singularity container
+A description of each image is beyond the scope of this document. 
+However, if you find an image useful for your application, you can use it by running an Apptainer container
  with the appropriate options. 
  For example:
 ```bash
-singularity run --nv --bind <bind_mount_path> /cvmfs/unpacked.cern.ch/<path_to_image>
+apptainer run --nv --bind <bind_mount_path> /cvmfs/unpacked.cern.ch/<path_to_image>
 ```
 
 #### Examples
 
-After installing package, you can then use GPU based machine learning algorithms. Two examples are supplied as an
-example.
-
-1. The first example aims at using a CNN to perform handwritten digits classification with `MNIST` dataset. The whole notebook can be found at [pytorch_mnist](../resources/gpu_resources/cms_resources/notebooks/pytorch_mnist.md). This example is modified from [an official `pytorch` example](https://github.com/pytorch/examples/tree/master/mnist).
-
-2. The second example is modified from the simple MLP example from [`weaver-benchmark`](https://github.com/colizz/weaver-benchmark). The whole notebook can be found at [toptagging_mlp](../resources/gpu_resources/cms_resources/notebooks/toptagging_mlp.md).
+After installing the package, you can then use GPU-based machine learning algorithms. Two examples are supplied.
 
+1. The first example aims at using a CNN to perform handwritten digits classification with `MNIST` dataset. The notebook can be found at [pytorch_mnist](../resources/gpu_resources/cms_resources/notebooks/pytorch_mnist.md). This example is modified from [an official `pytorch` example](https://github.com/pytorch/examples/tree/master/mnist).
 
+2. The second example is modified from the simple MLP example from [`weaver-benchmark`](https://github.com/colizz/weaver-benchmark). The notebook can be found at [toptagging_mlp](../resources/gpu_resources/cms_resources/notebooks/toptagging_mlp.md).
 
 ### Using Docker
-Docker is not supported at the moment in the interactive node of lxplus (like lxplus-gpu). However Docker is supported
+Docker is not currently supported in the interactive node of lxplus (like lxplus-gpu). However, Docker is supported
```

Comment:
Coming back to this. Now docker is directly supported in lxplus9 :)

Reply:
Yes, I noticed it's done through emulation of the CLI with Podman, which I think means there are some images that don't quite work out of the box (I had an issue with one)